### PR TITLE
feat(spanner): add samples for NUMERIC type

### PR DESCRIPTION
**Pull request by larkee**
Adds samples showing how to use the NUMERIC type in Cloud Spanner.

### CODE REVIEW

File: spanner/cloud-client/snippets.py [hunk 1]
```diff
@@ -1397,6 +1449,34 @@ def query_data_with_string(instance_id, database_id):
     # [END spanner_query_with_string_parameter]
 
 
+def query_data_with_numeric_parameter(instance_id, database_id):
+    """Queries sample data using SQL with a NUMERIC parameter. """
+    # [START spanner_query_with_numeric_parameter]
+    # instance_id = "your-spanner-instance"
```
Comment (hengfengli) [inline spanner/cloud-client/snippets.py:1455]: nits: should we add these comments for the above samples?

**Review by hengfengli — APPROVED** [review]
LGTM.

**Review by thiagotnunes — APPROVED** [review]
LGTM

File: spanner/cloud-client/snippets.py (hunk 1 above)
Comment (larkee) [reply spanner/cloud-client/snippets.py:1455]: I was matching similar samples. The difference seems to be that the region here begins inside the the function so the comments show that they are defined, whereas the other sample region tags after around the function which means the variables are defined as arguments.

We can consider making the region tags more consistent but that's out of scope for this PR

**Pull request merged**